File: Ruby/sort/bubble/1/main.rb
class Array
  def bubble_sort
    for i in 0...(size - 1)
      for j in 0...(size - i - 1)
        self[j], self[j + 1] = self[j + 1], self[j] if self[j] > self[j + 1]
      end
    end
    self
  end
end

puts [2, 1, 3, 5, 4].bubble_sort

File: Go/sort/bubble/1/main.go
package main

import (
	"sort"
	"fmt"
)

func main() {
	slice := sort.IntSlice{2, 1, 3, 5, 4}
	fmt.Println(bubbleSort(slice))
}

func bubbleSort(slice sort.Interface) sort.Interface {
	length := slice.Len()
	for i := 0; i < length - 1; i++ {
		for j := 0; j < length - 1 - i; j++ {
			if slice.Less(j+1, j) {
				slice.Swap(j, j+1)
			}
		}
	}
	return slice
}
